### PR TITLE
[Serving] Add logger prints

### PR DESCRIPTION
**Pull request by guy1992l**
Added logger prints to better understand in what phase of initialization a Nuclio v2 serving function get stuck when initialization fails.

### CODE REVIEW

**Review by liranbg — CHANGES_REQUESTED** [review]
Remove "..." that is not an interactive program 
Also, when logging with variables use
logger.debug_with("message", some="thing")

File: mlrun/serving/server.py [hunk 1]
```diff
@@ -306,18 +306,25 @@ def v2_serving_init(context, namespace=None):
...
     if config.log_level.lower() == "debug":
         server.verbose = True
     if hasattr(context, "trigger"):
         server.http_trigger = getattr(context.trigger, "kind", "http") == "http"
+    context.logger.info_with(
+        "Setting current function",
+        current_function=os.environ.get("SERVING_CURRENT_FUNCTION", ""),
+    )
     server.set_current_function(os.environ.get("SERVING_CURRENT_FUNCTION", ""))
+    context.logger.info("Initializing states")
```
Comment (liranbg) [inline mlrun/serving/server.py:320]:
```suggestion
    context.logger.info_with("Initializing states", namespace= namespace)
```

File: mlrun/serving/server.py [hunk 2]
```diff
@@ -306,18 +306,25 @@ def v2_serving_init(context, namespace=None):
...
     server.init_states(context, namespace or get_caller_globals())
+    context.logger.info("Initializing graph steps")
     serving_handler = server.init_object(namespace or get_caller_globals())
```
Comment (liranbg) [inline mlrun/serving/server.py:323]: i would suggest to also dive into `init_states` `init_object` and see if logs can be applied to them as well 
(as long as they r executed once, it is ok)

**Review by liranbg — CHANGES_REQUESTED** [review]
last one

**Comment by liranbg** [comment]
> Remove "..." that is not an interactive program Also, when logging with variables use logger.debug_with("message", some="thing")

note to self, serving function logger does not support log_with. its logger is mlrun logger. thus no need to do `x_with` but simply provide the variable. `logger.debug("message", some="thing")` @guy1992l fyi

**Review by liranbg — APPROVED** [review]

File: mlrun/serving/states.py [hunk 3]
```diff
@@ -828,6 +828,7 @@ def init_object(self, context, namespace, mode="sync", reset=False, **extra_kwar
 
         for step in self._steps.values():
             step.set_parent(self)
+            context.logger.info("Initializing step", step=step.name)
```
Comment (assaf758) [inline mlrun/serving/states.py:831]: This new log completely spams the log!
please remove @guy1992l @liranbg

**Pull request merged**